src/Microsoft.Azure.SignalR.Protocols/ServiceProtocol.cs

```diff
@@ -529,6 +566,30 @@ private static Claim[] ReadClaims(byte[] input, ref int offset)
             return null;
         }
 
+        private static Dictionary<string, StringValues> ReadHeaders(byte[] input, ref int offset)
+        {
+            var headerCount = ReadMapLength(input, ref offset, "headers");
+            if (headerCount > 0)
+            {
+                var headers = new Dictionary<string, StringValues>((int)headerCount);
```

Comment:
Shouldn't this be case insensitive? Do we have tests for that case?

Reply:
Yes, it should be case insensitive.
I have a question though. Since we just forward the headers from `HttpContext`, it really doesn't breka anything here, right? I assume headers with the same key have been aggregated by ASP.NET Core?

Reply:
Yes, it should be case insensitive.
I have a question though. Since we just forward the headers from `HttpContext`, it really doesn't break anything here, right? I assume headers with the same key have been aggregated by ASP.NET Core?

Reply:
It does, this code is in control of the creation of the `HttpContext`, there is no ASP.NET Core Server involved in the pipeline here. This basically replaces the server. So no, this logic is responsible for header aggregation.

Reply:
Are you talking about when service protocol is used out of Azure SignalR Service?
My understanding is when user's request is received by Azure SignalR Service runtime, ASP.NET Core has already handled the headers? Then headers are passed from SignalR Service to Server through `OpenConnectionMessage`. Am I missing anything?

Reply:
When using the SDK the http request is shuttled over the websocket connection via the service protocol. This logic needs to create a case insensitive dictionary.